linters/cpp/Rules.py

```diff
@@ -147,18 +148,45 @@ def namespace_check(ns_unified, full_path):
 
 	@staticmethod
 	def first_include_check(sorted_includes, path_elements):
-		del sorted_includes
+		filename_derived_header = '"{}.h"'.format(path_elements[-1][:-4])
+
 		full_path = '/'.join(path_elements)
 		if full_path in PLUGINS_FIRSTINCLUDES:
 			return '"{}"'.format(PLUGINS_FIRSTINCLUDES[full_path])
 
+		for sorted_include in sorted_includes:
+			if filename_derived_header == sorted_include.include:
+				return filename_derived_header
+
 		if 'validators' in path_elements:
 			return '"Validators.h"'
 
 		if 'observers' in path_elements:
 			return '"Observers.h"'
 
-		return '"{}.h"'.format(path_elements[-1][:-4])
+		return sorted_includes[0].include
+		# return '"{}.h"'.format(path_elements[-1][:-4])
+
+	@staticmethod
+	def _checked_o_or_v(path_elements):
```

Comment:
o or v means ?

Reply:
was for observers and validators, as I was adding this one by one, but it's perfectly ok to remove it now 🙃